cmd/sqlcmd/sqlcmd.go

```diff
@@ -117,25 +118,140 @@ func (a SQLCmdArguments) authenticationMethod(hasPassword bool) string {
+			}
+
+			if args.Help {
+				fmt.Print(cmd.UsageString())
+				os.Exit(0)
+			}
+
+			exitCode, err := run(vars, &args)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			fmt.Printf("after exit exitCode: %d\n", exitCode)
+			//os.Exit(exitCode)
+			fmt.Printf("after execute Function")
+		},
 	}
-	if args.Help {
-		_ = ctx.PrintUsage(false)
-		os.Exit(0)
+	setFlags(rootCmd, &args)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
 	}
-	vars := sqlcmd.InitializeVariables(!args.DisableCmdAndWarn)
-	setVars(vars, &args)
+}
+
+func setFlags(rootCmd *cobra.Command, args *SQLCmdArguments) {
+	var inputfiles []string
+	rootCmd.Flags().StringArrayVarP(&args.InputFile, "i", "i", inputfiles, "input file")
+	rootCmd.PersistentFlags().BoolVarP(&args.Version, "Version", "", false, "Print version information and exit")
+	rootCmd.PersistentFlags().StringVarP(&args.BatchTerminator, "Batch Terminator", "c", "GO", "Specifies the batch terminator. The default value is GO")
+	rootCmd.PersistentFlags().StringVarP(&args.UserName, "User Name", "U", "", "Specifies the batch terminator. The default value is GO")
+	rootCmd.PersistentFlags().StringVarP(&args.InitialQuery, "InitialQuery", "q", "", "Executes a query when sqlcmd starts, but does not exit sqlcmd when the query has finished running. Multiple-semicolon-delimited queries can be executed.")
+	rootCmd.PersistentFlags().StringVarP(&args.Query, "Query", "Q", "", "Executes a query when sqlcmd starts and then immediately exits sqlcmd. Multiple-semicolon-delimited queries can be executed.")
+	rootCmd.PersistentFlags().StringVarP(&args.Server, "Server", "S", "", "[tcp:]server[\\instance_name][,port]Specifies the instance of SQL Server to which to connect. It sets the sqlcmd scripting variable SQLCMDSERVER.")
+	rootCmd.PersistentFlags().BoolVarP(&args.DisableCmdAndWarn, "DisableCmdAndWarn", "X", false, "Disables commands that might compromise system security. Sqlcmd issues a warning and continues.")
+	rootCmd.PersistentFlags().StringVar(&args.AuthenticationMethod, "AuthenticationMethod", "", "Specifies the SQL authentication method to use to connect to Azure SQL Database. One of:ActiveDirectoryDefault,ActiveDirectoryIntegrated,ActiveDirectoryPassword,ActiveDirectoryInteractive,ActiveDirectoryManagedIdentity,ActiveDirectoryServicePrincipal,SqlPassword")
+	rootCmd.PersistentFlags().BoolVarP(&args.UseAad, "UseAad", "G", false, "Tells sqlcmd to use Active Directory authentication. If no user name is provided, authentication method ActiveDirectoryDefault is used. If a password is provided, ActiveDirectoryPassword is used. Otherwise ActiveDirectoryInteractive is used.")
+	rootCmd.PersistentFlags().BoolVarP(&args.DisableVariableSubstitution, "DisableVariableSubstitution", "x", false, "Causes sqlcmd to ignore scripting variables. This parameter is useful when a script contains many INSERT statements that may contain strings that have the same format as regular variables, such as $(variable_name).")
```

Comment:
>"DisableVariableSubstitution [](http://example.com/codeflow?start=71&length=28)

does Cobra automatically convert these strings to switches like --disable-variable-substitution?

Reply:
No , it converts the flag names to switches in the format '--flag-name'.
So here it would be '--DisableVariableSubstitution'. But we could override and provide custom name as the second parameter.
So i will override Name parameter by comparing with ODBC

Reply:
ODBC sqlcmd doesn't have any long names, just short flags.
Let's preserve the long names that the Kong implementation uses since it's already being used.